src/wallet/rpcwallet.cpp

```diff
@@ -3430,21 +3440,23 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
         }
     }
 
-    CBlockIndex *stopBlock = pwallet->ScanForWalletTransactions(pindexStart, pindexStop, true);
-    if (!stopBlock) {
-        if (pwallet->IsAbortingRescan()) {
-            throw JSONRPCError(RPC_MISC_ERROR, "Rescan aborted.");
+    CBlockIndex *stopBlock = nullptr;
+    {
```

Comment:
nit: why scope this? There's no lock to scope, could you not just leave this as-is and only change the chainActive.Tip() call?

Reply:
Your right,.. removed the unnecessary scope.